scripts/zones/The_Shrine_of_RuAvitau/npcs/qm2.lua

```diff
@@ -5,49 +5,25 @@
 -----------------------------------
 package.loaded["scripts/zones/The_Shrine_of_RuAvitau/TextIDs"] = nil;
 -----------------------------------
-
 require("scripts/zones/The_Shrine_of_RuAvitau/TextIDs");
-
------------------------------------
--- onTrade Action
------------------------------------
+require("scripts/zones/The_Shrine_of_RuAvitau/MobIDs");
+require("scripts/globals/status");
 
 function onTrade(player,npc,trade)
-    -- Validate traded items are all needed seals and ensure Kirin is not alive
-    if (GetMobAction(17506670) == 0 and trade:hasItemQty(1404, 1) and trade:hasItemQty(1405, 1) and trade:hasItemQty(1406, 1) and trade:hasItemQty(1407, 1) and trade:getItemCount() == 4) then
-        -- Complete the trade..
+    if (not GetMobByID(KIRIN):isSpawned() and trade:hasItemQty(1404, 1) and trade:hasItemQty(1405, 1) and trade:hasItemQty(1406, 1) and trade:hasItemQty(1407, 1) and trade:getItemCount() == 4) then
         player:tradeComplete();
-
-        -- Spawn Kirin..
-        local mob = SpawnMob(17506670, 180);
         player:showText(npc, KIRIN_OFFSET);
-        mob:updateClaim(player);
+        SpawnMob(KIRIN, 180):updateClaim(player);
```

Comment:
that 180 will make kirin instant despawn if claim is lost anywhere after 180 seconds. Surprised nobody corrected this yet. You want to remove the 180 and instead add the mobmod to the mob that we have specifically to control the despawn timer: the difference will be you will have 180 seconds from claim loss, instead of from mob spawn.